ksqldb-engine/src/main/java/io/confluent/ksql/engine/QueryMonitor.java

```diff
@@ -137,10 +151,12 @@ public void run() {
       LOG.info("KSQL query monitor started.");
 
       while (!closed) {
+        restartFailedQueries();
+
         try {
-          restartFailedQueries();
+          Thread.sleep(500);
```

Comment:
I added a few MS sleep just to not keep walking through the failed queries too often.